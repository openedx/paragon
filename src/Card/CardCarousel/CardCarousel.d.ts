import React from 'react';

export interface CardCarouselProps {
  ariaLabel: string,
  children: React.ReactNode,
  title?: React.ReactNode,
  subtitle?: React.ReactNode,
  columnSizes?: {
    xs?: number,
    sm?: number,
    md?: number,
    lg?: number,
    xl?: number,
  },
  hasInteractiveChildren?: boolean,
  canScrollHorizontal?: boolean,
  disableOpacityMasks?: boolean,
  onScrollPrevious?: Function,
  onScrollNext?: Function,
  CardCarouselControls?: React.ElementType,
}
declare const CardCarousel: React.ForwardRefExoticComponent<CardCarouselProps>;
export default CardCarousel;
