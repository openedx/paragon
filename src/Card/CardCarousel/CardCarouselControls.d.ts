import React from 'react';

declare const CardCarouselControls: React.FC;
export default CardCarouselControls;
