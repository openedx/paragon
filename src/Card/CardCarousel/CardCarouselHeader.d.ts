import React from 'react';

export interface CardCarouselHeaderProps {
  title?: React.ReactNode,
  subtitle?: React.ReactNode,
}
declare const CardCarouselHeader: React.FC<CardCarouselHeaderProps>;
export default CardCarouselHeader;
