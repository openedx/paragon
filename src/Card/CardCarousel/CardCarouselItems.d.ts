import React from 'react';

export interface CardCarouselItemsProps {
  children: React.ReactNode,
}
declare const CardCarouselItems: React.FC<CardCarouselItemsProps>;
export default CardCarouselItems;
