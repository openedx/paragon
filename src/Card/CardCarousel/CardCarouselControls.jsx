import React, { useContext } from 'react';
import { useIntl } from 'react-intl';
import Stack from '../../Stack';
import IconButton from '../../IconButton';
import Icon from '../../Icon';
import { ArrowForward, ArrowBack } from '../../../icons';
import { CardCarouselContext } from './CardCarouselProvider';
import messages from './messages';

function CardCarouselControls() {
  const {
    isScrolledToStart,
    isScrolledToEnd,
    scrollToPrevious,
    scrollToNext,
  } = useContext(CardCarouselContext);

  const intl = useIntl();

  if (isScrolledToStart && isScrolledToEnd) {
    return null;
  }

  return (
    <div className="pgn__card-carousel-controls">
      <Stack direction="horizontal" gap={1}>
        <IconButton
          disabled={isScrolledToStart}
          src={ArrowBack}
          iconAs={Icon}
          alt={intl.formatMessage(messages.scrollToPrevious)}
          onClick={scrollToPrevious}
        />
        <IconButton
          disabled={isScrolledToEnd}
          src={ArrowForward}
          iconAs={Icon}
          alt={intl.formatMessage(messages.scrollToNext)}
          onClick={scrollToNext}
        />
      </Stack>
    </div>
  );
}

export default CardCarouselControls;
