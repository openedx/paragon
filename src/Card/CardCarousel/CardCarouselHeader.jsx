import React, { useContext, isValidElement } from 'react';
import PropTypes from 'prop-types';
import CardCarouselTitle from './CardCarouselTitle';
import CardCarouselSubtitle from './CardCarouselSubtitle';
import DefaultCardCarouselControls from './CardCarouselControls';
import { CardCarouselContext } from './CardCarouselProvider';

const getFormattedTitle = (title) => {
  if (!title) {
    return null;
  }
  if (isValidElement(title)) {
    return title;
  }
  return <CardCarouselTitle>{title}</CardCarouselTitle>;
};

const getFormattedSubtitle = (subtitle) => {
  if (!subtitle) {
    return null;
  }
  if (isValidElement(subtitle)) {
    return subtitle;
  }
  return <CardCarouselSubtitle>{subtitle}</CardCarouselSubtitle>;
};

function CardCarouselHeader({ title, subtitle }) {
  const { CardCarouselControls: customCardCarouselControls } = useContext(CardCarouselContext);
  const CardCarouselControls = customCardCarouselControls || DefaultCardCarouselControls;

  const carouselTitle = getFormattedTitle(title);
  const carouselSubtitle = getFormattedSubtitle(subtitle);

  return (
    <div className="pgn__card-carousel-header">
      <div>
        {carouselTitle}
        {carouselSubtitle}
      </div>
      <CardCarouselControls />
    </div>
  );
}

CardCarouselHeader.propTypes = {
  title: PropTypes.node,
  subtitle: PropTypes.node,
};

CardCarouselHeader.defaultProps = {
  title: undefined,
  subtitle: undefined,
};

export default CardCarouselHeader;
