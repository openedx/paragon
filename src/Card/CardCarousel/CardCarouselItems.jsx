import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import CardDeck from '../CardDeck';
import { CardCarouselContext } from './CardCarouselProvider';

function CardCarouselItems({ children }) {
  const {
    columnSizes,
    hasInteractiveChildren,
    canScrollHorizontal,
    setOverflowRef,
  } = useContext(CardCarouselContext);

  return (
    <CardDeck
      ref={setOverflowRef}
      columnSizes={columnSizes}
      hasInteractiveChildren={hasInteractiveChildren}
      canScrollHorizontal={canScrollHorizontal}
      hasOverflowScrollItems
    >
      {children}
    </CardDeck>
  );
}

CardCarouselItems.propTypes = {
  children: PropTypes.node.isRequired,
};

export default CardCarouselItems;
