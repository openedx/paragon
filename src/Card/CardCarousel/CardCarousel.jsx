import React from 'react';
import PropTypes from 'prop-types';
import { OverflowScroll } from '../../OverflowScroll';
import CardCarouselProvider from './CardCarouselProvider';
import CardCarouselHeader from './CardCarouselHeader';
import CardCarouselItems from './CardCarouselItems';

function CardCarousel({
  ariaLabel,
  children,
  title,
  subtitle,
  columnSizes,
  hasInteractiveChildren,
  canScrollHorizontal,
  disableOpacityMasks,
  onScrollPrevious,
  onScrollNext,
  CardCarouselControls,
}) {
  return (
    <OverflowScroll
      disableScroll={!canScrollHorizontal}
      hasInteractiveChildren={hasInteractiveChildren}
      disableOpacityMasks={disableOpacityMasks}
      onScrollPrevious={onScrollPrevious}
      onScrollNext={onScrollNext}
      ariaLabel={ariaLabel}
    >
      <CardCarouselProvider
        columnSizes={columnSizes}
        hasInteractiveChildren={hasInteractiveChildren}
        canScrollHorizontal={canScrollHorizontal}
        CardCarouselControls={CardCarouselControls}
      >
        <div className="pgn__card-carousel">
          <CardCarouselHeader
            title={title}
            subtitle={subtitle}
          />
          <CardCarouselItems>{children}</CardCarouselItems>
        </div>
      </CardCarouselProvider>
    </OverflowScroll>
  );
}

CardCarousel.propTypes = {
  /** The `Card` items for the `CardCarousel`. */
  children: PropTypes.node.isRequired,
  /** Text describing the CardCarousel for screen readers */
  ariaLabel: PropTypes.string.isRequired,
  /** An optional title. */
  title: PropTypes.node,
  /** An optional subtitle. */
  subtitle: PropTypes.node,
  /** Customize the responsive columnSizes used by the carousel. */
  columnSizes: PropTypes.shape({
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
  }),
  /** Whether the carousel has interactive children (e.g., clickable cards). */
  hasInteractiveChildren: PropTypes.bool,
  /** Whether the carousel can be scrolled manually by users. */
  canScrollHorizontal: PropTypes.bool,
  /** Whether the default opacity masks should be shown at the start/end, if applicable. */
  disableOpacityMasks: PropTypes.bool,
  /** Callback function for when the user scrolls to the previous element. */
  onScrollPrevious: PropTypes.func,
  /** Callback function for when the user scrolls to the next element. */
  onScrollNext: PropTypes.func,
  /** Optional override for the default `CardCarouselControls` */
  CardCarouselControls: PropTypes.elementType,
};

CardCarousel.defaultProps = {
  title: undefined,
  subtitle: undefined,
  columnSizes: {
    sm: 12,
    lg: 6,
    xl: 4,
  },
  hasInteractiveChildren: false,
  canScrollHorizontal: true,
  disableOpacityMasks: false,
  onScrollPrevious: undefined,
  onScrollNext: undefined,
  CardCarouselControls: undefined,
};

export default CardCarousel;
