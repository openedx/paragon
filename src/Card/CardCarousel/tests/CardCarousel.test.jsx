import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import { v4 as uuidv4 } from 'uuid';
import { IntlProvider } from 'react-intl';
import CardCarousel from '../CardCarousel';
import Card from '../..';
import { OverflowScroll, OverflowScrollContext } from '../../../OverflowScroll';

jest.mock('../../../OverflowScroll', () => ({
  ...jest.requireActual('../../../OverflowScroll'),
  OverflowScroll: jest.fn(),
}));

// eslint-disable-next-line react/prop-types
OverflowScroll.mockImplementation(({ children }) => {
  const overflowRef = { current: document.createElement('div') };
  const contextValue = {
    overflowRef,
    isScrolledToStart: true,
    isScrolledToEnd: false,
    scrollToPrevious: jest.fn(),
    scrollToNext: jest.fn(),
  };

  return (
    <OverflowScrollContext.Provider value={contextValue}>
      <div data-testid="overflow-scroll">{children}</div>
    </OverflowScrollContext.Provider>
  );
});

function ExampleCard(props) {
  return (
    <Card {...props}>
      <Card.ImageCap src="http://fake.image" />
      <Card.Body>
        <Card.Header>Card title</Card.Header>
        <Card.Section>
          This is a wider card with supporting text below as a natural lead-in to
          additional content. This card has even longer content than the first to
          show that equal height action.
        </Card.Section>
      </Card.Body>
      <Card.Footer textElement={<small className="text-muted">Last updated 3 mins ago</small>} />
    </Card>
  );
}

function getCardContent({ cardCount = 5, ...cardProps } = {}) {
  return Array.from({ length: cardCount }).map(() => <ExampleCard key={uuidv4()} {...cardProps} />);
}

function CardCarouselWrapper({ children, ...rest }) {
  return (
    <IntlProvider locale="en">
      <CardCarousel {...rest}>
        {children}
      </CardCarousel>
    </IntlProvider>
  );
}

CardCarouselWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  ariaLabel: PropTypes.string,
};

CardCarouselWrapper.defaultProps = {
  ariaLabel: 'example card carousel',
};

describe('<CardCarousel />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders default card carousel', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardCarouselWrapper>
        {cardContent}
      </CardCarouselWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders card carousel with title and subtitles', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardCarouselWrapper
        title="Example Title"
        subtitle="Example subtitle."
      >
        {cardContent}
      </CardCarouselWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders card carousel with custom title and subtitles elements', () => {
    const cardContent = getCardContent();
    const tree = renderer.create((
      <CardCarouselWrapper
        title={<h1>Example Title</h1>}
        subtitle={<p className="small">Example subtitle</p>}
      >
        {cardContent}
      </CardCarouselWrapper>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
