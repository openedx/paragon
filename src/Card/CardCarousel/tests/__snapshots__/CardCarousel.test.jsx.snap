// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCarousel /> renders card carousel with custom title and subtitles elements 1`] = `
<div
  data-testid="overflow-scroll"
>
  <div
    className="pgn__card-carousel"
  >
    <div
      className="pgn__card-carousel-header"
    >
      <div>
        <h1>
          Example Title
        </h1>
        <p
          className="small"
        >
          Example subtitle
        </p>
      </div>
      <div
        className="pgn__card-carousel-controls"
      >
        <div
          className="pgn__hstack pgn__stack-gap--1"
        >
          <button
            aria-label="Scroll to previous"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={true}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            aria-label="Scroll to next"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={false}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
    >
      <div
        className="pgn__card-deck-row row"
        tabIndex={0}
      >
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardCarousel /> renders card carousel with title and subtitles 1`] = `
<div
  data-testid="overflow-scroll"
>
  <div
    className="pgn__card-carousel"
  >
    <div
      className="pgn__card-carousel-header"
    >
      <div>
        <h2
          className="pgn__card-carousel-title"
        >
          Example Title
        </h2>
        <p
          className="pgn__card-carousel-subtitle"
        >
          Example subtitle.
        </p>
      </div>
      <div
        className="pgn__card-carousel-controls"
      >
        <div
          className="pgn__hstack pgn__stack-gap--1"
        >
          <button
            aria-label="Scroll to previous"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={true}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            aria-label="Scroll to next"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={false}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
    >
      <div
        className="pgn__card-deck-row row"
        tabIndex={0}
      >
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CardCarousel /> renders default card carousel 1`] = `
<div
  data-testid="overflow-scroll"
>
  <div
    className="pgn__card-carousel"
  >
    <div
      className="pgn__card-carousel-header"
    >
      <div />
      <div
        className="pgn__card-carousel-controls"
      >
        <div
          className="pgn__hstack pgn__stack-gap--1"
        >
          <button
            aria-label="Scroll to previous"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={true}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
          <button
            aria-label="Scroll to next"
            className="btn-icon btn-icon-primary btn-icon-md"
            disabled={false}
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="btn-icon__icon-container"
            >
              <span
                className="pgn__icon btn-icon__icon"
              >
                <svg
                  aria-hidden={true}
                  fill="none"
                  focusable={false}
                  height={24}
                  role="img"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="pgn__card-deck pgn__card-deck-has-horizontal-scroll"
    >
      <div
        className="pgn__card-deck-row row"
        tabIndex={0}
      >
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
        <div
          className="pgn__overflow-scroll-item pgn__card-deck__card-item col-xl-4 col-lg-6 col-sm-12"
        >
          <div
            className="pgn__card pgn__card-light card"
            tabIndex={-1}
          >
            <div
              className="pgn__card-wrapper-image-cap vertical"
            >
              <img
                className="pgn__card-image-cap"
                onError={[Function]}
                onLoad={[Function]}
                src="http://fake.image"
              />
            </div>
            <div
              className="pgn__card-body"
            >
              <div
                className="pgn__card-header"
              >
                <div
                  className="pgn__card-header-content"
                />
              </div>
              <div
                className="pgn__card-section"
              >
                This is a wider card with supporting text below as a natural lead-in to additional content. This card has even longer content than the first to show that equal height action.
              </div>
            </div>
            <div
              className="pgn__card-footer vertical"
            >
              <div
                className="pgn__card-footer-text vertical"
              >
                <small
                  className="text-muted"
                >
                  Last updated 3 mins ago
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
